clap_complete/src/shells/fish.rs

```diff
@@ -20,8 +20,38 @@ impl Generator for Fish {
             .get_bin_name()
             .expect("crate::generate should have set the bin_name");
 
+        let has_global_flags = {
+            let n_args = cmd.get_arguments().filter(|a| !a.is_positional()).count();
+            // `-h` and `-v` behaves as command, not regular flags/options that modifies subcommand behaviors.
+            match (
+                cmd.is_disable_help_flag_set(),
+                cmd.is_disable_version_flag_set(),
+            ) {
+                (false, false) => n_args > 2,
+                (true, true) => n_args > 0,
+                _ => n_args > 1,
+            }
```

Comment:
If its the behavior you are interested in checking for, you likely want to look to see what `ArgAction` is set.